src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -32593,9 +32703,9 @@ void M3([MaybeNull]out string x)
 
     void M4([NotNull]out string? x)
     {
-        x = null; // 2
-        (x, _) = (null, 1); // 3
-    }
+        x = null;
+        (x, _) = (null, 1);
+    } // 2
```

Comment:
hmm. this makes it start to feel like a tradeoff for more complex methods. Imagine a complex method with many return points and many writes to some `[NotNull] out` param. Would you feel lost if the nullability warnings were on the returns instead of the original assignments?

Are we also trying to improve the experience in scenarios where users write a value that might be null to the out param, and if it is null, they write some fallback value that's known to be non-null?

Reply:
Yes, there is a bit of trade-off. Warning sooner points to the location of the problematic assignment, but it also means that we flag temporary assignments (annoying). 
For nullability feature in general, we lean away from anything that could be annoying and could hold you back from annotating your correct code as it already exists.